lib/shared/sdk/usbboot/index.js

```diff
@@ -260,35 +296,38 @@ const startFileServer = (device, endpoint, files) => {
       return Bluebird.try(() => {
         if (fileMessage.command === protocol.FILE_MESSAGE_COMMANDS.GET_FILE_SIZE) {
           debug(`Getting the size of ${fileMessage.fileName}`)
-          const fileSize = _.get(files, [ fileMessage.fileName, 'length' ])
 
-          if (_.isNil(fileSize)) {
+          return queryBlobFromCache(fileMessage.fileName, readFile).then((fileBuffer) => {
```

Comment:
Oh yeah, I'd forgotten that the usbboot protocol requests the filesize first - that's another very good reason to support file-caching :)